kod/object/passive/itematt/weapatt/wavamper.kod

```diff
@@ -58,31 +57,37 @@ messages:
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
       
    ModifyDamage(damage = 0, target = $, wielder = $, lData= $)
-   "Wielder has a 1% chance of stealing life from target with any hit."
+   "Wielder has a 10% chance of stealing life from target with any hit."
    {
+      local iStolen;
+
       if random(1,100) <= piEffect_percent
       {
-	 send(wielder,@GainHealth,#amount = bound(damage,0,8));
-    	 send(wielder, @MsgSendUser, #message_rsc = vamper_worked,
-	    #parm1=send(target,@getcapdef),#parm2=send(target,@GetName),
-	    #parm3=send(target,@GetHisHer));
-    	 send(target, @MsgSendUser, #message_rsc = vamper_worked_target,
-	    #parm1=send(wielder,@getcapdef),#parm2=send(wielder,@GetName),
-	    #parm3=send(wielder,@GetHisHer));
-
-
+         iStolen = bound(damage,0,8);
+         iStolen = bound(iStolen,0,Send(target,@GetHealth)-1);
+         iStolen = Send(target,@AssessDamage,#what=wielder,#damage=iStolen,
```

Comment:
ModifyDamage is just supposed to return a modified damage number.  Turning one attack into two like this seems very strange and possibly quite unsafe, since it bypasses any other modifiers on the original attack.  I would have expected to see just ```damage = damage + iStolen;``` here.

Reply:
@akirmse Are you okay with the damage not matching the life that is taken? I made this change on my test, armor and resist can lower how much damage the target takes, but I cannot tell later how much came from the addition so the wielder will always heal 8

Reply:
Interesting. Perhaps there should be a smaller life steal more frequently, if we can't take armor and resistances into account.

Reply:
Ok I added the damage to the attack.